var/spack/repos/builtin/packages/oce/package.py

```diff
@@ -87,7 +87,11 @@ def install(self, spec, prefix):
                 '-DOCE_OSX_USE_COCOA:BOOL=ON',
             ])
 
-        options.append('-DCMAKE_INSTALL_NAME_DIR:PATH=%s/lib' % prefix)
+        if '.'.join(platform.mac_ver()[0].split('.')[:2]) == '10.12':
+            # use @rpath on Sierra due to limit of dynamic loader
+            options.append('-DCMAKE_MACOSX_RPATH=ON')
```

Comment:
@tgamblin in real life, most projects consist of only a handful of libraries, so the current standard arguments in `CMakePackage` are fine. To my knowledge it's only `Trilinos` and `oce` which have hundreds of libraries which may really break things on Sierra.